tests/testthat/test-build_factory.R

```diff
@@ -200,3 +191,24 @@ test_that("Can build factories that branch into functions by variable.", {
   #   dpi = 96
   # )
 })
+
+
+test_that("Factories can have state", {
+  new_counter <- build_factory(
```

Comment:
This test failed for me. Did something you changed while trying to refactor kill it?

Reply:
Nope...all the tests pass for me.

Reply:
I added an additional scenario to the `.state` test around passing a state that needs to be evaluated in the calling environment.  In earlier builds, the state list was being captured with `enexpr` and this would cause passed variables to be quoted, rather than evaluated.

Reply:
my bad.  last commit wasn't pushed.